### DISCUSSION TITLE
중급프로젝트 2주차 팀회고

### DISCUSSION THREAD

**Discussion by yeji0512** [codeit-bootcamp-frontend/2-Weekly-Mission#430] [회고]
중급 프로젝트 2주차 팀 회고를 하고([회고 방식 상세보기](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/discussions/429#discussion-6130547)), 이전처럼 댓글 남겨주세요. 
이번 팀 회고에서 점검해보면 좋을 것 같은 내용이예요. 

- 프로젝트 2주차이지만 여전히 정리 되지 않은 팀 규칙, 컨벤션이 있다면 이번 시간에 정리해보는 것도 좋을 것 같아요. 남은 프로젝트 기간에 더 좋은 협업을 할 수 있을거예요. 
- 팀 소통 방식 
- 협업 관계에서 느낀 감정 
- 기타 등등 

![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/144666341/c3c205ca-e63a-4791-9b16-13874b276af2)

**Comment by 9ummy9** [comment]
**7팀**
---

**Keep** 👌
   - 지난 회고의 Try 사항들이 잘 개선됐다.
   - 계획한 작업 일정이 잘 지켜지고 있다.
   - 다들 팀미팅 시간 및 코어 타임에 성실히 참여했다.

**Problem** 😢
   - 대시보드와 헤더를 합친 레이아웃을 실제 적용해보니 사용에 어려움이 있어서 코드를 수정해야 했다.
   - 코드 리뷰를 좀 더 꼼꼼히 했으면 좋겠다.
   - 팀원 간 의사소통 시 서로 이해한 바가 달라지지 않게 좀 더 명확한 의견 전달이 필요하다.
   - 추후 수정하지 않도록 API 문서를 잘 확인하고 컴포넌트를 만들었으면 좋겠다.
  
**Try** 👍
   - 본인의 작업 사항 중 헷갈리는 점이 있다면 팀원들에게 정확히 확인하고 진행하는 것이 좋을 것 같다. 

<br>

<img width="385" alt="스크린샷_2024-02-01_123055-removebg-preview" src="https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148167964/69049d90-0305-48ca-9959-03ad6e571861">


7팀 파이팅!

**Comment by mizign** [comment]
## Keep
- 원활한 커뮤니케이션
- 다들 적극적으로 질문함 
- 트러블 슈팅 지속적으로 기록했다
- 깃 브랜칭 전략에 다들 익숙해졌다!

## Problem
- 최적화를 신경쓰지 못한 코드 작성 (오버페칭 등..) 
- 서로 PR 확인하고 머지하기
- 스웨거 잘 확인하고 작업 시작하기 

## Try
- PR 확인했으면 본문에 이모지 남기고, 마지막 사람이 머지하기
- 최적화하기 
- 로딩, 에러 처리하기

**Comment by Chyyni** [comment]
## 4팀

### keep
- 매일 회의 때 정해진 루틴(템플릿)이 있어서 상황 공유가 잘 되는 것 같다.
- 노션에 회의록, 트러블 슈팅, 공유 할 정보 등 작업에 관련해서 자세하게 기록하는게 좋다.
- 다들 코어 타임에 잘 참여하고 의사소통이 잘 이루어지고 있다.
- 문제를 서로 공유하고 함께 고민하는 시간을 가져, 문제 해결 시간을 단축할 수 있었다.


### problem
- PR 확인이 첫 주보다 느리고 리뷰가 적어졌다.
- pr에 명확하고 간결한 정보 작성이 어려웠다.
- pr을 올리지 않고 바로 main에 merge하는 일이 생긴다.
- 작업 속도가 느리고 디테일한 부분까지 신경을 못쓰고 있는 것 같다.
- 맡은 작업의 진행 속도가 팀에 비해 느리다
- 페이지 별로 나누다보니 겹치는 부분이 있었는데 초기에 파악하지 못하고 각자 진행하여 나중에 합치려다보니 힘들었다
- git 사용법이 아직 어려운 것 같다.


### try
- PR 확인 적극적으로, 리뷰 달기
- PR 제대로 올리기
- 잘 안 풀리는 문제 적극적으로 질문하기 / 혼자 너무 오래 고민하지 않기
- 겹치는 부분이 있을경우 초기에  대화를 해서 나누거나 페어프로그래밍을 진행해야 할 것 같다
- git에 익숙해져 충돌을 잘 해결할 수 있도록 해야겠다
- pr, merge 이후에 보이는 문제(변수명, 로직, ui 등)를 사전에 더 잘 체크할 필요가 있다.
- 작업 속도 상승을 위해 더 노력할 필요가 있다.
- 머지하기 전에 한번더 꼼꼼하게 확인해야겠다.
- 원활한 작업을 위해, 부족한 부분 공부를 더 해야할 것 같다.

**Comment by grapefruit13** [comment]
# 8팀

## KEEP
- PR 리뷰 :  덕분에 코드 컨벤션을 지킬 수 있고, 코드 가독성을 높일 수 있어서 좋습니다. 내가 고쳐야 할 코드가 무엇인지 알 수 있고 더 나은 코드를 참고할 수 있습니다.
- 코어타임 : 다들 계속 온라인인 덕분에 의사소통이 빠르게 이루어져서 좋습니다.
- 브랜치 관리 : 프로젝트 내의 파트를 세분화하여 브랜치를 관리하기 때문에, 상황 파악 및 코드 관리가 용이합니다.
- 노션 정리 : 중요 사항들을 노션에 문서화하기 때문에, 팀원들간의 내용 공유 및 숙지가 편합니다.
- 스크럼 :  팀원들의 현재 상황을 알 수 있어 남은 과제를 어떤 순서로 해야 할 지 정할 수 있습니다.

## PROBLEM
- 스크럼 시간이 필요 이상으로 길어집니다. 오늘도 1시간 반정도 했습니다.
- 바로 dev에 머지가 되지 않는 다는 점 때문에 PR을 다소 가볍게 생각하고 올리는 경우가 있는 것 같습니다. 
- 파일 경로, 파일명같이 중요한 부분은 꼭 인지하고  PR 올렸으면 좋겠습니다.
- 다른 팀원들의 피드백은 까먹지 않게 적어두고, 꼼꼼히 반영하면 좋겠습니다.
- dev에 머지한 후에 컴포넌트가 의도와 다르게 작동하거나 보이는 경우가 많습니다.
- 노션 정리를 특정 팀원만 하는 것 같아서 아쉽습니다.
- 버그 때문에 너무 많은 시간이 소요됩니다. 얘기하면 수정 안 돼 있어서 두세 번 말하다 보면 시간 금방 갑니다.
- 본인 부분 수정 얘기가 나오면 적어야 합니다. (스크럼 끝나고 또 물어봄)
- 의견을 많이 나눠야 하는데 다양한 의견이 나오지 않습니다.

## TRY
- 스크럼에서는 핵심적이고 중요한 이슈를 얘기하고, 가벼운 이슈는 채팅이나 팀원들의 소규모 통화로 하면 좋겠습니다. 시간은 최대 한 시간으로 했으면 좋겠습니다.
- 다른 팀원들의 리뷰만으로는 모든 버그를 방지할 수 없기 때문에,  보다 꼼꼼히 테스트하고 확인한 후 PR을 올리면 좋겠습니다.
- 당장 해결하지 못하는 버그들은 무조건 github에 issue로 올려서 추후 고칠 수 있도록 했으면 좋겠습니다.
- 해결한 issue는 바로바로 close 해주어, 프로젝트 상황을 쉽게 파악할 수 있도록 해주세요.
- 질문이나 의견을 얘기할 때는, 다른 팀원이 한 번 더 물을 필요가 없도록 구체적이고 명확하게 표현해주세요.
- 이제 시간이 일주일도 안 남은 만큼 더 효율적으로 협업할 수 있으면 좋겠습니다.
- 다들 더 적극적으로 노션에 기록했으면 좋겠습니다. 특히 문제해결 기록!
- 프로젝트에서 모르는 부분이 있다면 그냥 넘기지 말고, 그것에 대해 제대로 찾아봐서 지식을 습득했으면 합니다.
- 남이 짠 코드도 최대한 들여다 봐야 할거같아요!
- PR이나 commit을 하기 전 규칙에 따라서 작성했는지 확인하고, 작성한 코드가 컨벤션을 따르고 있는지 한 번 더 확인하며, 수정해야 할 부분을 완벽하게 수정했는지 확인합니다.

**Comment by muyahho** [comment]
![스크린샷 2024-02-05 011210](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/88885994/2a8a5d1e-0245-4000-8820-74a9df24dc39)

**Comment by bumsly** [comment]
![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/65000254/34cf27cf-d26c-4b09-ba0e-0aaa8e6ef041)

**Reply by bumsly** [reply]
https://www.figma.com/file/a1fmZRmIgWfJ2vEygGnu5a/Untitled?type=design&node-id=0-1&mode=design&t=23ePKulrCvol6tvE-0

**Comment by eonpain** [comment]
# 5팀

## KEEP
- 1주차 회고할 때 상태가 그대로입니다. 소통 문제, 마일스톤 내에 계획한 걸 지키는 것의 문제도 여전합니다.

## PROBLEM
- 1주차와 변함이 없습니다. pr을 서로 확인하며 얻는 자극같은 긍정적인 영향도 팀원 일부만 받는 것 같습니다.
- 물리적, 심리적으로 프로젝트 진행하는데 좀 버거움이 있습니다.

## TRY
- 기간 내게 잘 마무리할 수 있게 서로 조금만 더 힘냈으면 합니다. 5팀.. 화이팅!

**Comment by rladngus133** [comment]
## Keep 👌
- 팀미팅을 오래하며 의사소통이 잘 되어있다.
- 각자 짠 코드의 이해성을위하여 사용법이 필요한코드의경우 문서화를 철저하게하여 협업의 능률이 올라갔다.
- 한 명도 지각없이 일정을 잘 지켜주었다!
- GitHub의 Project로 워크플로우를 관리하고있어 보다 GitHub의 기능을 더 알게되었으며 각자의 일정관리가 편해졌다.

## Problem 😢
- 동일기능의 함수를 중복되게 만드는 문제가있어 조금더 세밀한 소통이 필요 할 것 같다.

## Try 👍
- 스스로 만들며 중복 될 것 같으면 팀내소통을 통하여 빠르게 해결하는방향으로 개선하였다!

9팀파이팅 ~ !
![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/124851297/a1014e9e-8f21-4174-8ab4-b36b3c30284c)
_API 만드는 9팀모습_

**Comment by ynmkim** [comment]
## 10팀

### **Keep**

- 진행 상황 및 이슈만 빠르고 짧게 공유하는 스크럼이 좋습니다. 개인 작업 시간에 조금 더 집중 할 수 있습니다.
- 새로운 기술들에 대해서 다양하게 이야기를 나누고, 시도해 볼 수 있는 개방적인 분위기가 좋습니다.
- 프로젝트 또한 학습의 과정으로 존중해주고 동기 부여 해주는 점이 좋습니다.

### Problem

- 아직까지 큰 이슈없이 순조롭게 진행되고 있습니다.

### Try

- 리팩토링 시간이 되다면 멘토님이 피드백 주신 내용을 개선해보면 좋을 것 같습니다. 1) 서로의 코드가 다른데 통일하려고 노력해보기 2) 서로 리뷰하기 (분해하는 듯이 분석하기)

**Comment by juncastle97** [comment]
# 3팀 회고
## keep
- 다들 맡은 부분을 잘 해주고 있습니다.
- 피드백이 빠릅니다!
- 의견 반영을 잘 해줍니다.
- 결정하는 거에 있어서 시간 낭비를 하지 않고 빠르게 결정됩니다.
## problem
- 단위 통일, 폴더 이름 잘 쓰기, 함수 이름 잘 쓰기 등 컨벤션 잘 지키기
- 질문 많이 하기
## try
- 한숨 10번 이상 하지 않기
- 컴포넌트로 분리 할 수 있는 거 분리하기
- 정신차리기 (얼마 안 남았습니다!)

3팀 파이팅😀

**Comment by sung-eunji** [comment]
# 2팀 회고
## keep
리뷰가 전보다 활발해졌다! 감사하다!
리뷰를 꼼꼼히 해주셔서 배워가는 점이 많네요
디코에서 PR이랑 머지할 때마다 잘 알려주셔서 좋았습니다.
'pr - merge' 과정에 규칙이 정해져서 깃 순환이 잘 됩니다.
비교적 자유로운 분위기: 서로가 조심스럽지만 솔직하게 상태를 공유하고 있는 분위기가 좋습니다.
회의록작성 해야 할일에 대한 구분이 명확해 작업대기부분에서 찾아서 작업하면 되는부분이 좋다.
본인의 상황 진도 상황에 대한 공유가 확실함

## problem
생산성이 좀 떨어지는 것 같다.
코드 일관성이 떨어지는 것 같다.
주로 하는 사람만 리뷰 하므로 안 하시는 분도 해주셨으면 좋다.
컴포넌트 작성시 서로 작성하는 스타일에 대한 차이 부분 (이벤트 처리 prop 상태 관리에 대한) 의견 부족
맡은 작업의 진행 속도가 팀에 비해 느리다.

## try
질문을 편하게 하자! 멘토님께도 여쭙자!
코드 컨벤션(네이밍, 타입 등?)을 더 명확히 하자(여력이 되면..)
우리... 코어 타임에는 같이 음성 채팅방에 있을래요...? 같이 진행상황 실시간으로 얘기하며 작업해요...
압박 웰컴입니다 ..!
PR 확인 적극적으로, 리뷰 달기하겠습니다.
원활한 작업을 위해, 부족한 부분 공부를 더 해야할 것 같다.


![lol](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/116065415/46040339-57e6-4608-87b0-37706e859949)
